File: src/main/java/com/rabbit/api/dto/request/LoginRequest.java
package com.rabbit.api.dto.request;

import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class LoginRequest {

    @NotBlank(message = "Email or username is required")
    private String identifier;

    @NotBlank(message = "Password is required")
    private String password;
}

File: dtos/requests/LoginRequest.js
const { BadRequest } = require('../../middleware/errorHandler');

class LoginRequest {
  constructor({ identifier, password }) {
    if (!identifier) throw new BadRequest('identifier is required');
    if (!password) throw new BadRequest('password is required');

    this.identifier = identifier;
    this.password = password;
  }
}
module.exports = LoginRequest;
